core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -0,0 +1,350 @@
+    def testProduceNeedsAuthorization() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Describe)), topicResource)
+      try {
+        sendRecords(numRecords, tp)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: ApiException => Assert.assertEquals(Errors.AUTHORIZATION_FAILED.exception().getMessage, e.getMessage)
+      }
+    }
+
+    @Test
+    def testOnlyWritePermissionAllowsWritingToProducer() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      sendRecords(numRecords, tp)
+    }
+
+    @Test
+    def testCreatePermissionNeededForWritingToNonExistentTopic() {
+      val newTopic = "newTopic"
+      val topicPartition = new TopicPartition(newTopic, 0)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), new Resource(Topic, newTopic))
+      try {
+        sendRecords(numRecords, topicPartition)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: TimeoutException =>
+        //TODO Need to update the producer so it actually throws the server side of exception.
+        case e: Exception => Assert.fail("Only timeout exception should be thrown.")
+      }
+
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)), Resource.ClusterResource)
+      sendRecords(numRecords, topicPartition)
+    }
+
+    @Test
+    def testConsumerNeedsAuthorization() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      //TODO: Ideally we would want to test that when consumerGroup permission is not present we still get an AuthorizationException
+      //but the consumer fetcher currently waits forever for the consumer metadata to become available.
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+      sendRecords(1, tp)
+      try {
+        this.consumers.head.assign(List(tp).asJava)
+        consumeRecords(this.consumers.head)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: AuthorizationException => Assert.assertEquals("Not authorized to read from topic-0", e.getMessage)
+      }
+    }
+
+    @Test
+    def testAllowingReadOnTopicAndGroupAllowsReading() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), topicResource)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+      sendRecords(1, tp)
+      this.consumers.head.assign(List(tp).asJava)
+      consumeRecords(this.consumers.head)
+    }
+  
+//    TODO: The following test goes into an infinite loop as consumer waits for consumer metadata to be propogated for ever.
```

Comment:
This fix was needed in #366 also. If that gets in first, then you don't need to change it here.